### PR TITLE
add support for remote command actions

### PR DESCRIPTION
**Pull request by joelddiaz**
no support for anything but custom scripts at this time

### CODE REVIEW

File: roles/lib_zabbix/library/zbx_action.py [hunk 1]
```diff
@@ -301,7 +347,41 @@ def get_operation_type(inc_operation):
 
     return o_types[inc_operation]
 
-def get_action_operations(zapi, inc_operations):
+def action_remote_command(ansible_module, zapi, operation):
+    ''' Process remote command type of actions '''
+
+    if str(operation['opcommand']['type']) == "0": # 0 = custom script
```
Comment (twiest) [inline roles/lib_zabbix/library/zbx_action.py]: I would prefer something like this:

```
CUSTOM_SCRIPT_ACTION="0"

if str(operation['opcommand']['type']) == CUSTOM_SCRIPT_ACTION:
```

File: roles/lib_zabbix/library/zbx_action.py [hunk 2]
```diff
@@ -301,7 +347,41 @@ def get_operation_type(inc_operation):
...
+
+        # If 'execute_on' wasn't defined, assume we're going to run directly
+        # on zabbix server
+        if 'execute_on' not in operation['opcommand']:
+            operation['opcommand']['execute_on'] = 1 # 1 = zabbix server
```
Comment (twiest) [inline roles/lib_zabbix/library/zbx_action.py]: same

File: roles/lib_zabbix/library/zbx_action.py [hunk 3]
```diff
@@ -315,9 +395,14 @@ def get_action_operations(zapi, inc_operations):
             else:
                 operation['opmessage']['default_msg'] = 0
 
-        # NOT supported for remote commands
-        elif operation['operationtype'] == 1:
-            continue
+        elif operation['operationtype'] == 1: # remote command
```
Comment (twiest) [inline roles/lib_zabbix/library/zbx_action.py]: same

File: roles/lib_zabbix/library/zbx_action.py [hunk 4]
```diff
@@ -315,9 +395,14 @@ def get_action_operations(zapi, inc_operations):
...
+            #
+            # Currently only implmented support for "custom script" type
+            #
+            if 'type' not in operation['opcommand']:
+                operation['opcommand']['type'] = "0" # 0 = custom script by default
```
Comment (twiest) [inline roles/lib_zabbix/library/zbx_action.py]: same

**Comment by kwoodson** [comment]
:+1:

File: roles/lib_zabbix/library/zbx_action.py [hunk 5]
```diff
@@ -315,9 +401,16 @@ def get_action_operations(zapi, inc_operations):
             else:
                 operation['opmessage']['default_msg'] = 0
 
-        # NOT supported for remote commands
-        elif operation['operationtype'] == 1:
-            continue
+        #elif operation['operationtype'] == 1: # remote command
```
Comment (twiest) [inline roles/lib_zabbix/library/zbx_action.py]: remove.

**Comment by joelddiaz** [comment]
@twiest @kwoodson removed the "set this to a default if not provided" behavior. Trying to follow whether zabbix has a default or not (https://www.zabbix.com/documentation/2.4/manual/api/reference/action/object#action)

**Comment by twiest** [comment]
@joelddiaz great job with this one.

:+1:

**Pull request merged**